ktlint-ruleset-standard/src/main/kotlin/com/pinterest/ktlint/ruleset/standard/rules/ChainMethodContinuation.kt

```diff
@@ -0,0 +1,113 @@
+import com.pinterest.ktlint.ruleset.standard.StandardRule
+import org.jetbrains.kotlin.com.intellij.lang.ASTNode
+import org.jetbrains.kotlin.com.intellij.psi.tree.TokenSet
+
+/**
+ * When present, align chain operators('.' or '?.') at previous line with next method call
+ */
+public class ChainMethodContinuation :
+    StandardRule("chain-method-continuation"),
+    Rule.Experimental {
+    private val chainOperators =
+        TokenSet.create(
+            ElementType.DOT_QUALIFIED_EXPRESSION,
+            ElementType.SAFE_ACCESS_EXPRESSION,
+        )
+
+    override fun beforeVisitChildNodes(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        node.takeIf {
+            it.elementType in chainOperators
+        }?.let {
+            visitChains(node, autoCorrect, emit)
+        }
+    }
+
+    private fun visitChains(
+        node: ASTNode,
+        autoCorrect: Boolean,
+        emit: (offset: Int, errorMessage: String, canBeAutoCorrected: Boolean) -> Unit,
+    ) {
+        val chainOperator = node.findChildByType(ElementType.SAFE_ACCESS) ?: node.findChildByType(ElementType.DOT)
+        chainOperator ?: return
+        val rightExpression = chainOperator.nextCodeSibling()?.firstChildLeafOrSelf() ?: return
+        val leftExpressionEndBrace =
+            chainOperator.prevLeaf { !it.isWhiteSpace() && it.elementType != ElementType.EXCLEXCL }
+                ?.lastChildLeafOrSelf() ?: return
+        val leftExpressionBeforeEndBrace = leftExpressionEndBrace.prevCodeSibling()?.lastChildLeafOrSelf() ?: return
+        val isLeftExpressionEndBraceInSeparateLine =
+            leftExpressionEndBrace.elementType == ElementType.RBRACE &&
+                hasNewLineInClosedRange(leftExpressionBeforeEndBrace, leftExpressionEndBrace)
+        if (!noNewLineInOpenRange(chainOperator, rightExpression)) {
+            emit(
+                chainOperator.textRange.endOffset,
+                "${
+                    if (isLeftExpressionEndBraceInSeparateLine) {
+                        leftExpressionEndBrace.leavesIncludingSelf()
+                            .takeWhile { it != chainOperator && !it.isWhiteSpace() }.map { it.text }
+                            .joinToString(separator = "")
+                    } else {
+                        ""
+                    }
+                }${chainOperator.text} must merge at the start of next call",
+                true,
+            )
+            if (autoCorrect) {
+                if (isLeftExpressionEndBraceInSeparateLine) {
+                    /*
+                        Detects code like below
+                        bar {
+                            ...
+                        }.
+                        foo() // this should align with previous line }
+                     */
+                    chainOperator.upsertWhitespaceAfterMe("")
```

Comment:
This leaves an empty PsiWhitespace elements in the AST. Technically that is not a problem. But I think that removing an unwanted whitespace element is more clear.

Reply:
Updated with `nextMultilineWhiteSpace.treeParent.removeChild(nextMultilineWhiteSpace)`